ledger-tool/src/main.rs

```diff
@@ -388,6 +390,76 @@ fn graph_forks(
     dot.join("\n")
 }
 
+macro_rules! analyze_column {
+    ($db:expr, $col:ident, $name:expr, $ksize:expr) => {
+        let mut key_hist = histogram::Histogram::new();
+        let mut key_tot: u64 = 0;
+        let mut val_hist = histogram::Histogram::new();
+        let mut val_tot: u64 = 0;
+        let mut row_hist = histogram::Histogram::new();
+        let a = $ksize as u64;
+        for (_x, y) in $db.iter::<$col>(blocktree_db::IteratorMode::Start).unwrap() {
+            let b = y.len() as u64;
+            key_hist.increment(a).unwrap();
+            key_tot += a;
+            val_hist.increment(b).unwrap();
+            val_tot += b;
+            row_hist.increment(a + b).unwrap();
+        };
+        println!("{}", json!({
+            "column":$name,
+            "entries":key_hist.entries(),
+            "key_stats":{
+                "max":key_hist.maximum().unwrap(),
+                "total_bytes":key_tot,
+            },
+            "val_stats":{
+                "p50":val_hist.percentile(50.0).unwrap(),
+                "p90":val_hist.percentile(90.0).unwrap(),
+                "p99":val_hist.percentile(99.0).unwrap(),
+                "p999":val_hist.percentile(99.9).unwrap(),
+                "min":val_hist.minimum().unwrap(),
+                "max":val_hist.maximum().unwrap(),
+                "stddev":val_hist.stddev().unwrap(),
+                "total_bytes":val_tot,
+            },
+            "row_stats":{
+                "p50":row_hist.percentile(50.0).unwrap(),
+                "p90":row_hist.percentile(90.0).unwrap(),
+                "p99":row_hist.percentile(99.0).unwrap(),
+                "p999":row_hist.percentile(99.9).unwrap(),
+                "min":row_hist.minimum().unwrap(),
+                "max":row_hist.maximum().unwrap(),
+                "stddev":row_hist.stddev().unwrap(),
+                "total_bytes":key_tot + val_tot,
+            },
+        }).to_string());
```

Comment:
try `to_string_pretty()` so the output isn't all bunched up into one hard-to-read line